### PR TITLE
Add wait_for_rgw in test scripts

### PR DESCRIPTION
**Pull request by hemanthnakkina**
# Description

Add function wait_for_rgw in test scripts.
The function takes number of daemons as
parameter and checks against ceph status.

## Type of change

Please delete options that are not relevant.

- [x] CleanCode (Code refactor, test updates, does not introduce functional changes)

## Contributor's Checklist

Please check that you have:

- [x] self-reviewed the code in this PR.

### CODE REVIEW

**Comment by hemanthnakkina** [comment]
This patch is getting tested here - https://github.com/canonical/charm-microceph/pull/83

File: tests/scripts/actionutils.sh [hunk 1]
```diff
@@ -322,6 +312,26 @@ function wait_for_osds() {
     fi    
 }
 
+function wait_for_rgw() {
+    local expect="${1?missing}"
+    res=0
+    for i in $(seq 1 8); do
+	res=$( ( sudo microceph.ceph -s | grep -F rgw: | sed -E "s/.* ([[:digit:]]*) daemon.*/\1/" ) || true )
+        if [[ $res -ge $expect ]] ; then
+            echo "Found ${expect} rgw daemon"
```
Comment (sabaini) [inline tests/scripts/actionutils.sh:321]: Minor nit: suggest "Found ${expect} rgw daemon(s)"

File: tests/scripts/actionutils.sh [hunk 2]
```diff
@@ -322,6 +312,26 @@ function wait_for_osds() {
...
+            break
+        else
+            echo -n '.'
+            sleep 5
+        fi
+    done
+    sudo microceph.ceph -s
+    if [[ $res -lt $expect ]] ; then
+        echo "Never reached ${expect} rgw"
```
Comment (sabaini) [inline tests/scripts/actionutils.sh:330]: Minor nit: suggest "Never reached ${expect} rgw daemon(s)"

**Review by sabaini — CHANGES_REQUESTED** [review]
Thanks @hemanthnakkina, this looks useful -- 2 minor wording nits

**Comment by hemanthnakkina** [comment]
> This patch is getting tested here - [canonical/charm-microceph#83](https://github.com/canonical/charm-microceph/pull/83)

Tested succsfully against charm-microceph PR: https://github.com/canonical/charm-microceph/actions/runs/9413791300/job/25931789677?pr=83

**Review by sabaini — APPROVED** [review]
Cheers

**Pull request merged**